src/proto/single/mod.rs

```diff
@@ -1,77 +1,122 @@
 use chrono::{DateTime, Utc};
+use derive_builder::Builder;
 use std::collections::HashMap;
 use std::io::prelude::*;
 
 mod cmd;
 mod resp;
+mod utils;
 
-use crate::error::Error;
+use crate::error::{self, Error};
 
 pub use self::cmd::*;
 pub use self::resp::*;
 
+const JOB_DEFAULT_QUEUE: &str = "default";
+const JOB_DEFAULT_RESERVED_FOR_SECS: usize = 600;
+const JOB_DEFAULT_RETRY_COUNT: usize = 25;
+const JOB_PRIORITY_MAX: u8 = 9;
+const JOB_DEFAULT_PRIORITY: u8 = 5;
+const JOB_DEFAULT_BACKTRACE: usize = 0;
+
 /// A Faktory job.
 ///
+/// To create a job, use 'Job::new' specifying 'kind' and 'args':
+/// ```
+/// use faktory::Job;
+///
+/// let _job = Job::new("order", vec!["ISBN-13:9781718501850"]);
+/// ```
+///
+/// Alternatively, use 'JobBuilder' to construct a job:
+/// ```
+/// use faktory::JobBuilder;
+///
+/// let result = JobBuilder::default()
+///     .kind("order")
+///     .args(vec!["ISBN-13:9781718501850"])
+///     .build();
+/// if result.is_err() {
+///     todo!("Handle me gracefully, please.")
+/// };
+/// let _job = result.unwrap();
+/// ```
+///
 /// See also the [Faktory wiki](https://github.com/contribsys/faktory/wiki/The-Job-Payload).
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, Builder)]
+#[builder(
+    setter(into),
+    build_fn(name = "try_build", private, validate = "Self::validate")
+)]
 pub struct Job {
     /// The job's unique identifier.
+    #[builder(default = "utils::gen_random_jid()")]
     pub(crate) jid: String,
 
     /// The queue this job belongs to. Usually `default`.
+    #[builder(default = "JOB_DEFAULT_QUEUE.into()")]
     pub queue: String,
 
     /// The job's type. Called `kind` because `type` is reserved.
     #[serde(rename = "jobtype")]
     pub(crate) kind: String,
```

Comment:
Given that `kind` is a required argument, I'd rather it have to be passed to `builder`. That seems to be the generally-recommended pattern for options that must be set for the build to succeed. Look at the doc code elsewhere, that doesn't seem to be the case at the moment?

Reply:
An alternative if the derive crate doesn't let you do that is to have it generate a private constructor (i.e., to make `JobBuilder` not implement `Default` ), and instead have a `Job::builder(kind)` that returns a `JobBuilder` where the `kind` has been set.

Reply:
This 'default' in current implementation is definitely deceptive: as if it were enough to build a simple job, while it's not and it really should be followed/enriched by `.kind()`  and `.args()` , otherwise the error will be returned.

With these updates:

1)  as per your suggestion, the required arg is set of the JobBuilder right away (either via JobBuilder::new("jobtype" or  - also added as an alternative way to instantiate a builder - via Job::builder("jobtype")));

2) the 'args' field has been given a default of 'Vec::new()' on the JobBuilder;

3) as per your suggestions, we got rid of the 'priority' validation (As a side-note: on the second thought, I realized that adding a 'JobPriority' (e.g. JobPriority::from(3)) new type would have been a better solution compared to erroring back at a job's build time. In the new type case they would be able to set/reset the priority as many time as they want and it would still be valid. But this 1) looks like a subject of a different PR and 2) not necessary in the first place);

We can:
1. Create a new Job with JobBuilder infallibly;

2. Get rid of the 'Client' error variant (I am using this one in another branch on tracking job's progress for Faktory enterprise, BUT I am using it there in the exact same case: checking that 'percent' value in the range from 0 to 100. I now think I should be adding this validation there to not constrain the user and plus no instructions on this are given in Faktory docs).